app.py

```diff
@@ -98,26 +97,83 @@ def find_font(name):
 videos = get_videos(searchPath)
 video_names = get_video_names(searchPath)
 
-# Preload the response; it's always the same
-subs = subs2json(load_subs(videos))
+# Loads the video subtitles
+video_subs = load_subs(videos)
+# Creates a list of video data with subtitles
+videos_with_subs = [{ 'title': video_names[idx], 'id': idx, 'subs': v[0] } for idx, v in enumerate(video_subs)]
+# Convert sub data to JSON data
+subs = subs2json(video_subs)
+# Creates a list of the installed fonts
+fonts = font_manager.findSystemFonts(fontpaths=None, fontext='ttf')
 
 # Find a suitable font
 font = find_font('Noto')
 
+# Jinja2 templates
+templates = Environment(loader = FileSystemLoader('templates'))
+
 logger.info("ready to start the application")
 
+# Serves anything stored in public folder
+@app.route("/public/<path:path>")
+def get_public(path):
+    return send_from_directory("public", path)
+
+# Shows the main UI
+@app.route("/")
+def get_index():
+    search = request.args.get("q")
+    if search is not None:
+        filtered_subs = [{ **video, 'subs': [sub for sub in video['subs'] if search.lower() in sub.text] } for video in videos_with_subs]
+        return render_template("sublist.html", videos=filtered_subs)
```

Comment:
Although I did indeed not implement it in this PR, pagination needs to be a serious consideration here. Especially if you have many videos, showing all the subtitles of all the videos will make the page incredibly slow. The idea here, and in fact I can implement it, is to not show all subtitles in one big list, but instead have pages (perhaps one page per episode) with the subtitles, and if you then search, it makes sense for the backend to do that work, because the backend is the single source of truth, instead of having to duplicate the state (i.e. all subtitles) on both the backend and frontend

Reply:
Also one word: caching. The browser will cache previously searched queries this way